api/circulation.py

```diff
@@ -342,6 +342,15 @@ def borrow(self, patron, pin, licensepool, delivery_mechanism,
             )
 
         new_loan = False
+
+        # TODO: If the book isn't available and the patron is putting it on hold,
+        # we shouldn't check the loan limit. Or should there be one limit for loans
+        # and holds?
+        loan_limit = patron.library.setting(Configuration.LOAN_LIMIT).int_value
+        non_open_access_loans = [loan for loan in patron.loans if loan.license_pool.open_access == False]
+        if loan_limit and len(non_open_access_loans) >= loan_limit:
+            raise PatronLoanLimitReached()
```

Comment:
We want to count loans towards the limit only if your loan is interfering with someone else's ability to read the book. So open access books shouldn't count, but neither should books obtained through OPDS For Distributors. I don't have a good idea for determining whether a book's loan should 'count' apart from checking how the loan was set up.

Reply:
That's a good point. How about checking if the loan has an end date?

Reply:
Yeah, that seems pretty reliable.